### ISSUE TITLE
Error parsing multipart/form-data response 

### ISSUE THREAD

**Issue by tomasbonnedahl** [http4k/http4k#327]
Receiving the following headers and body as a response, I'm not able to parse the parts. Exception thrown:

`org.http4k.multipart.ParseError: no name for part`

Headers received:
```
HTTP/1.1 200 OK
Date: Fri, 13 Dec 2019 09:10:19 GMT
Content-Type: multipart/form-data; boundary=ofz0Oi4n1E68F6IC91sf36bcHAX5sy
Transfer-Encoding: chunked
Connection: keep-alive
Y-Tro-Optapi-Requestid: ofz0Oi4n1E68F6IC91sf36bcHAX5sy
Strict-Transport-Security: max-age=15724800; includeSubDomains
Server: trorisalve
```

Body received:
```
--ofz0Oi4n1E68F6IC91sf36bcHAX5sy
Content-Disposition: attachment; name=values
Content-Type: text/csv

A,B,C
932,059A,2019-11-25
933,96A,2019-11-25


--ofz0Oi4n1E68F6IC91sf36bcHAX5sy
Content-Disposition: attachment; name=logs
Content-Type: application/x-ndjson

{"level":"info","time":1576656181033,"message":"Starting, build 01e3342d 2019-12-11 12:03:51 +0100 (master) with run id ","pid":152568,"thread_id":152568}
{"level":"info","time":1576656181033,"message":"Reading input from stdin...","pid":152568,"thread_id":152568}

--ofz0Oi4n1E68F6IC91sf36bcHAX5sy--

```
The exception is thrown in `StreamingMultipartFormsPart.kt`:
```
val contentDisposition = ParameterParser().parse(headers["Content-Disposition"], ';')
val fieldName = (if (contentDisposition.containsKey("attachment")) mixedName else trim(contentDisposition["name"]))
                ?: throw ParseError("no name for part")
```
`mixedName` is `null`, and should probably only be used within a `multipart/mixed` context if I read the code correctly?

Call site code:
```
val client = ApacheClient()
val response = client(request)
val multipart = MultipartFormBody.from(response)
println("multipart boundary = ${multipart.boundary}")
```

http4k version: 3.196.0

**Comment by daviddenton** [comment]
Yes - this is because multipart Response aren't supported. There is a way around it though. Something like this is *really* hacky, but should work: 

https://gist.github.com/daviddenton/e809a4e188a818256d995e4e7ffde220

**Comment by tomasbonnedahl** [comment]
Thanks @daviddenton. Tried the hack but got stuck in `Response.Companion.parse` where the lines are being split by CRLF and then joined with "" resulting in only one row without line feed characters.

**Comment by tomasbonnedahl** [comment]
If someone else stumbles across the same issue, I ended up using https://github.com/synchronoss/nio-multipart which seems to work out of the box.

**Comment by alphaho** [comment]
I think I could take a look at this issue

**Comment by alphaho** [comment]
To add the multipart response support, should I create a set of classes and lenses similar to `MultipartFormBody` , `MultipartEntity` ,  `MultipartFormParser` , `Part`  etc.?
If so, is it possible to reuse some of the existing logic? Because the current `http4k-multipart` module seems very specific to `multipart/form-data` requests, it feels not very straight-forward to reuse.

Or should I “fix” the hacky example instead? (I think the workaround is hacky is because it added the http status to each of the parts so that we can reuse `Response.Companion::parse` ’s parsing logic. Is it correct?)

**Comment by daviddenton** [comment]
I don’t see the hack as a viable forward option at all. I also know that the current implementation of multipart form data is pretty complicated. 

At the very least I’d probably duplicate the required logic to start with and we can see where we go from there. It might be that we can tease out some common code a bit later.

**Comment by alphaho** [comment]
I see. Then I will try to come up with something similar to the existing multipart form data setup.
